### PR TITLE
added DL verification

### PR DESCRIPTION
**Pull request by asariev**

### CODE REVIEW

File: src/main/java/io/onroad/eventconsumer/service/validator/CreateUserRequestValidator.java [hunk 1]
```diff
@@ -15,13 +16,19 @@
...
   public void validate(User loggedInUser, CreateUserRequest request) {
 
     if (request == null || loggedInUser == null) {
       throw new ValidationException(ValidationExceptionType.BAD_REQUEST);
     }
 
+    if(request.getLicense() != null && driverLicenseRepo.findByDriverLicenseNumber(
+            request.getLicense().getDriverLicenseNumber()).isPresent()){
+      throw new ValidationException(ValidationExceptionType.USER_ALREADY_CREATED_BY_DRIVER_LICENSE);
+    }
```
Comment (onroadeld) [inline src/main/java/io/onroad/eventconsumer/service/validator/CreateUserRequestValidator.java:30]: According to  ELD-FUNC 4.1.2.4 it should be on per motor carrier (company) level, no? So basically if driver changes his motor carrier he should be able to register again on our system with a different company?

File: src/main/java/io/onroad/eventconsumer/domain/model/DriversLicense.java [hunk 2]
```diff
@@ -20,10 +22,11 @@
   private Long id;
 
   @NotNull
-  private String number;
+  private String driverLicenseNumber;
```
Comment (onroadeld) [inline src/main/java/io/onroad/eventconsumer/domain/model/DriversLicense.java:25]: I think we can omit `driverLicense` part here as the class name already says that it's DriversLicense, no?

Current -> `DriversLicense.number`
New      -> `DriversLicense.driversLicenseNumber`

File: src/main/java/io/onroad/eventconsumer/domain/model/DriversLicense.java [hunk 3]
```diff
@@ -20,10 +22,11 @@
...
 
   @NotNull
-  private String issuer;
+  @Size(max = 2)
+  private String issuedState; // For UI, it should be dropdown, select state code
```
Comment (onroadeld) [inline src/main/java/io/onroad/eventconsumer/domain/model/DriversLicense.java:29]: How about using enum and allowing only those they are truly valid?

File: src/main/java/io/onroad/eventconsumer/service/UserService.java [hunk 4]
```diff
@@ -31,20 +33,22 @@
...
   private final CreateUserRequestValidator createUserRequestValidator;
+  private final GeneralValidator generalValidator;
 
   @Transactional
   public UserView create(User loggedInUser, CreateUserRequest request) {
 
+    if(request.getLicense() != null) {
+      request.getLicense().setDriverLicenseNumber(
+              generalValidator.safeString(request.getLicense().getDriverLicenseNumber().toUpperCase()));
+    }
```
Comment (onroadeld) [inline src/main/java/io/onroad/eventconsumer/service/UserService.java:44]: Why we are changing the value here, maybe returning 400 and asking for a valid number with error message would be better?

File: src/main/resources/application.properties [hunk 5]
```diff
@@ -1,6 +1,6 @@
 spring.jpa.database-platform=org.hibernate.dialect.MySQL5InnoDBDialect
 spring.jpa.generate-ddl=false
-spring.jpa.hibernate.ddl-auto=none
+spring.jpa.hibernate.ddl-auto=create
```
Comment (onroadeld) [inline src/main/resources/application.properties:3]: We have `schema.sql` class which is used to create ddl, no?

**Review by onroadeld — CHANGES_REQUESTED** [review]
ELD-FUNC-4.1.2.2 says drivers license and issued state should be required when creating DRIVER accounts. Should we make it a required and validate accordingly?

File: src/main/java/io/onroad/eventconsumer/domain/model/DriversLicense.java (hunk 2 above)
Comment (asariev) [reply src/main/java/io/onroad/eventconsumer/domain/model/DriversLicense.java:25]: we can leave as its then

File: src/main/java/io/onroad/eventconsumer/domain/model/DriversLicense.java (hunk 3 above)
Comment (asariev) [reply src/main/java/io/onroad/eventconsumer/domain/model/DriversLicense.java:29]: Baary State code enum kylyp kirgizip koioiunbu?

File: src/main/java/io/onroad/eventconsumer/service/UserService.java (hunk 4 above)
Comment (asariev) [reply src/main/java/io/onroad/eventconsumer/service/UserService.java:44]: In DL, format is XXXX-XXX-XXXX, but in report we should generate without dash, I was thinking just removed the extra special characters. Or from FrontEnd we need to specify the format

File: src/main/resources/application.properties (hunk 5 above)
Comment (asariev) [reply src/main/resources/application.properties:3]: Mende run bolboitat, super user'di da manual run kylyp alamyn

File: src/main/java/io/onroad/eventconsumer/domain/model/DriversLicense.java (hunk 3 above)
Comment (onroadeld) [reply src/main/java/io/onroad/eventconsumer/domain/model/DriversLicense.java:29]: ooba

File: src/main/java/io/onroad/eventconsumer/service/UserService.java (hunk 4 above)
Comment (onroadeld) [reply src/main/java/io/onroad/eventconsumer/service/UserService.java:44]: If its only formatting difference then it should be handled on the frontend and frontend when saving should send only the data.

File: src/main/resources/application.properties (hunk 5 above)
Comment (onroadeld) [reply src/main/resources/application.properties:3]: Emne kata chygyp jatat?

Comment (asariev) [reply src/main/resources/application.properties:3]: For running application, its giving me error that I have a new column. For test case's, schema is running first
should we add here:

`
@SpringBootApplication
public class EventConsumerApplication {

`

Comment (asariev) [reply src/main/resources/application.properties:3]: Ech kandai error chykpaile

File: src/main/java/io/onroad/eventconsumer/service/validator/GeneralValidator.java [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Component;
+
+@Component
+@RequiredArgsConstructor
+public class GeneralValidator {
+
+    public String safeString (String str){
+        return str.replaceAll("[^a-zA-Z0-9]+","");
+    }
```
Comment (onroadeld) [inline src/main/java/io/onroad/eventconsumer/service/validator/GeneralValidator.java:12]: Let's delete this class.

**Review by onroadeld — COMMENTED** [review]
Left tiny comment, all looks good .

**Review by onroadeld — APPROVED** [review]

**Pull request merged**